.github/workflows/build.yml

```diff
@@ -28,136 +28,107 @@ jobs:
+              with:
+                  node-version: 18
+            - uses: actions/cache/restore@v4
+              with:
+                  path: ./
+                  key: web3-18-${{ github.sha }}
+            - run: yarn build:esm
+
     build-types:
         name: Build Types
         needs: build
         runs-on: ubuntu-latest
         steps:
-            -   uses: actions/setup-node@v4
-                with:
-                    node-version: 18
-            -   uses: actions/cache/restore@v4
-                with:
-                    path: ./
-                    key: web3-18-${{github.sha}}
-            -   run: yarn build:types
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: 18
+            - uses: actions/cache/restore@v4
+              with:
+                  path: ./
+                  key: web3-18-${{ github.sha }}
+            - run: yarn build:types
+
     lint:
         name: Lint
         needs: build
         runs-on: ubuntu-latest
         steps:
-            -   uses: actions/setup-node@v4
-                with:
-                    node-version: 18
-            -   uses: actions/cache/restore@v4
-                with:
-                    path: ./
-                    key: web3-18-${{github.sha}}
-            -   run: npx ts-node scripts/init.ts
-            -   name: Restore eslint caches
-                uses: actions/cache/restore@v4
-                with:
-                    path: |
-                        packages/web3/.eslintcache
-                        packages/web3-core/.eslintcache
-                        packages/web3-eth/.eslintcache
-                        packages/web3-eth-abi/.eslintcache
-                        packages/web3-eth-accounts/.eslintcache
-                        packages/web3-eth-contract/.eslintcache
-                        packages/web3-eth-ens/.eslintcache
-                        packages/web3-eth-iban/.eslintcache
-                        packages/web3-eth-personal/.eslintcache
-                        packages/web3-net/.eslintcache
-                        packages/web3-providers-http/.eslintcache
-                        packages/web3-providers-ws/.eslintcache
-                        packages/web3-rpc-methods/.eslintcache
-                        packages/web3-types/.eslintcache
-                        packages/web3-utils/.eslintcache
-                        packages/web3-validator/.eslintcache
-                        tools/web3-plugin-example/.eslintcache
-                    key: ${{ runner.os }}-eslintcache
-
-            -   run: yarn lint
-            -   run: gh cache delete "${{ runner.os }}-eslintcache"
-                if: github.event_name == 'push' && github.ref == 'refs/heads/4.x'
-                env:
-                    GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-            -   name: Save eslint caches
-                if: github.event_name == 'push' && github.ref == 'refs/heads/4.x'
```

Comment:
After further consideration, I realized this cache setup is unnecessary since the cache is only being deleted and saved on the 4.x branch. By removing this extra logic, we allow actions/cache@v4 to manage the cache automatically—saving it when needed and restoring it if available. Please review the recent builds on this branch for reference.

Reply:
I finally understand what was happening here. The workflow was set up to store the lint cache for the 4.x branch and restore it for PR branches to lint only the diffs in a PR. The issue was that the 4.x cache didn’t exist, likely due to the cache reaching the 10GB limit, which resulted in older caches being deleted.